elasticapm/utils/compat.py

```diff
@@ -107,7 +107,26 @@ def iteritems(d, **kwargs):
     def iterlists(d, **kw):
         return d.iterlists(**kw)
 
-
+    def exec_(_code_, _globs_=None, _locs_=None):
+        """Execute code in a namespace."""
+        if _globs_ is None:
+            frame = sys._getframe(1)
+            _globs_ = frame.f_globals
+            if _locs_ is None:
+                _locs_ = frame.f_locals
+            del frame
+        elif _locs_ is None:
+            _locs_ = _globs_
+        exec("""exec _code_ in _globs_, _locs_""")
+
+    exec_(
+        """def reraise(tp, value, tb=None):
+    try:
+        raise tp, value, tb
+    finally:
+        tb = None
+"""
+    )
```

Comment:
My brain really dislikes this. Any reason we're not using `six` instead?

Reply:
Depending on six is somewhat problematic, as it is a very widely used dependency. Meaning we can't require any specific version of six, as that might clash with requirements of another module that depends on six.

We used to vendor the complete six module, but at some point removed the vast majority of it because we only use a very small subset. The result is the `elasticapm.utils.compat` module.

We can totally discuss if declaring six as a dependency would be the better approach, but if it's ok with you, let's do that in a different issue.

Reply:
Honestly this is likely what `six` is doing under the covers anyway, so I'm fine. I assumed there was a reason but was just curious. :)

Reply:
This is a straight copy from `six` :) there's a copyright note in the source, so it should be kosher